Python/bltinmodule.c

```diff
@@ -2549,8 +2549,8 @@ zip_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
         PyObject *it = PyObject_GetIter(item);
         if (it == NULL) {
             if (PyErr_ExceptionMatches(PyExc_TypeError))
```

Comment:
Might make sense to do the same for all exceptions types, but I'm not sure if it's OK for compatibility.